packages/cryptography/src/Ed25519PublicKey.js

```
@@ -106,6 +111,25 @@ export default class Ed25519PublicKey extends Key {
         return Ed25519PublicKey.fromBytes(hex.decode(text));
     }
 
+    /**
+     * @param {Uint8Array} data
+     * @returns {boolean}
+     */
+    static isValid(data) {
+        if (!data) return false;
+
+        const EMPTY_BUFFER = Buffer.from(new Uint8Array(33).fill(0));
+        const bufferData = Buffer.from(data);
+        if (bufferData.equals(EMPTY_BUFFER)) return true;
+
+        try {
+            ed25519.ExtendedPoint.fromHex(data);
+            return true;
+        } catch {
+            return false;
```

If an error is caught wouldn't it make sense to propagate it.

---

I think considering the method's name is `isValidKey` you would expect it to return `true` or `false`.

---

throw new BadKeyError("invalid public key"); is thrown in another layer. The message of the error for the key parse would be useful there imho. I know you've defined it so but that error could get up there somehow

---

Since this method is created in this PR, we can ditch it and do the validation in the `fromBytesRaw` func. There we can `throw new BadKeyError("invalid public key: ", e);`, giving more info to the user as Georgi said.